File: C/Partie4/Exo14.2.c
#include <stdio.h>
#include <unistd.h>
#include <stdlib.h>
#include <string.h>
#include <sys/wait.h>

int main(int argc, char *argv[])
{
    int tube[2];
    char buffer[512];

    tube[0] = atoi(argv[1]);
    tube[1] = atoi(argv[2]);

    close(tube[1]);

    read(tube[0], buffer, 512);
    printf("\nMessage recu: %s\n", buffer);
    close(tube[0]);

    return 0;
}

File: Python/Partie4/Exo14.2.py
import os
import sys

if len(sys.argv) != 3:
    sys.exit(1)
fd_lecture = int(sys.argv[1])
fd_ecriture = int(sys.argv[2])
os.close(fd_ecriture)
data = os.read(fd_lecture, 256)
print("Message reçu:", data.decode())